Bundle/app/src/main/java/com/scottlindley/Bundle/ContentDBHelper.java

```diff
@@ -0,0 +1,346 @@
+        if(removed < 1) {
+            //
+        }
+    }
+
+    /**
+     * Insert a new row into the tweets table
+     * @param db
+     * @param handle
+     * @param tweet
+     * @param username
+     * @param time
+     * @param id
+     * @return The row number that was inserted, or -1 if nothing was inserted
+     */
+    private long addTweet(SQLiteDatabase db, String handle, String tweet, String username, String time, String id){
+        ContentValues values = new ContentValues();
+        values.put(COL_NAME, handle);
+        values.put(COL_CONTENT, tweet);
+        values.put(COL_USERNAME, username);
+        values.put(COL_TIME, time);
+        values.put(COL_TWEET_ID, id);
+        return db.insert(TABLE_TWEETS, null, values);
+    }
+
+    private long addNews(SQLiteDatabase db, String title, String summary, String link) {
+        ContentValues values = new ContentValues();
+        values.put(COL_NAME, title);
+        values.put(COL_CONTENT, summary);
+        values.put(COL_URL, link);
+        return db.insert(TABLE_NEWS, null, values);
+    }
+
+    private List<CardContent> combineResults(List<CardContent> cards, Cursor tweetCursor, Cursor newsCursor) {
+        int count = 1;
+        if(tweetCursor.moveToFirst()&&newsCursor.moveToFirst()) {
+            while(!tweetCursor.isAfterLast()) {
+                if(count%3==0) {
+                    String title = newsCursor.getString(newsCursor.getColumnIndex(COL_NAME));
+                    String summary = newsCursor.getString(newsCursor.getColumnIndex(COL_CONTENT));
+                    String link = newsCursor.getString(newsCursor.getColumnIndex(COL_URL));
+                    cards.add(new NewsStory(title, summary, link));
+                    newsCursor.moveToNext();
+                    count++;
+                } else {
+                    String handle = tweetCursor.getString(tweetCursor.getColumnIndex(COL_NAME));
+                    String tweet = tweetCursor.getString(tweetCursor.getColumnIndex(COL_CONTENT));
+                    String username = tweetCursor.getString(tweetCursor.getColumnIndex(COL_USERNAME));
+                    String time = tweetCursor.getString(tweetCursor.getColumnIndex(COL_TIME));
+                    String id = tweetCursor.getString(tweetCursor.getColumnIndex(COL_TWEET_ID));
+                    cards.add(new TweetInfo(handle, tweet, username, time, id));
+                    tweetCursor.moveToNext();
+                    count++;
+                }
+            }
+        }
+        tweetCursor.close();
+        newsCursor.close();
+        return cards;
+    }
+
+    private void startRefreshService() {
+        JobScheduler jobScheduler = (JobScheduler)mContext.getSystemService(Context.JOB_SCHEDULER_SERVICE);
+
+        JobInfo newsInfo = new JobInfo.Builder(NEWS_SERVICE, new ComponentName(mContext, NewsService.class))
+                .setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY)
+                .build();
```

Comment:
I don't like the idea of triggering network requests from inside the DB handler. Obviously it works, but in terms of "separation of concerns" this makes the code difficult to follow and maintain. The database should be able to receive data and store it, but should not be in charge of going out and getting the data.